pkg/gui/recent_repos_panel.go

```diff
@@ -1,24 +1,57 @@
 package gui
 
 import (
+	"fmt"
 	"os"
 	"path/filepath"
+	"strings"
+	"sync"
 
 	"github.com/jesseduffield/generics/slices"
 	"github.com/jesseduffield/lazygit/pkg/commands"
 	"github.com/jesseduffield/lazygit/pkg/env"
+	"github.com/jesseduffield/lazygit/pkg/gui/presentation/icons"
 	"github.com/jesseduffield/lazygit/pkg/gui/style"
 	"github.com/jesseduffield/lazygit/pkg/gui/types"
 )
 
+func (gui *Gui) getCurrentBranch(path string) string {
+	if branch, err := gui.os.Cmd.New(
+		fmt.Sprintf("git -C %s rev-parse --abbrev-ref HEAD", gui.os.Quote(path)),
+	).RunWithOutput(); err == nil {
+		return strings.Trim(branch, "\n")
+	}
+	return ""
+}
+
 func (gui *Gui) handleCreateRecentReposMenu() error {
 	recentRepoPaths := gui.c.GetAppState().RecentRepos
 
+	currentBranches := sync.Map{}
```

Comment:
that's one hell of an improvement! Nice work. Nonetheless, it still takes about a second on my end for 77 repos. Given that people typically expect menus to open instantly for the sake of muscle-memory (e.g. I might go to the status panel, hit enter then immediately hit enter again to go to the previous repo), I reckon we need to find a way to first render the menu without branches, then after it's been rendered, re-render it with the branches. I admittedly don't know how to do this off the top of my head but for all we know it's very straight-forward. I can dig into this deeper if you want.

Reply:
> Nonetheless, it still takes about a second on my end for 77 repos.

I'm really curious as to why is this the case. Is it always the same repo that's causing the issue or is something else amidst? 

> I reckon we need to find a way to first render the menu without branches, then after it's been rendered, re-render it with the branches.

That might look a bit janky if the branch name is the middle column, since we don't know the exact width of the longest branch, unless the second render shows both the branches and the paths.

> I admittedly don't know how to do this off the top of my head but for all we know it's very straight-forward. **I can dig into this deeper if you want.**

I'd like to take a stab at it first then if I don't manage to find anything out I'll tug at your sleeve, deal? :)

Reply:
I'm not sure which repo is the culprit, but even if it is a random repo causing the issue, I reckon it's better to just assume that some users are going to have their repo branches load slower than others, so we should try to make it smooth for everyone.

It might look a bit janky but I think it should be fine: users are typically going to be looking at the left column for the actual repo name so at least that will remain unchanged.

> deal? :)

Deal :)

Reply:
> we should try to make it smooth for everyone.

Of course, couldn't agree more, I was just curious because I'm stumped, is all :smile: